src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3382,6 +3388,61 @@ class Config:
         await self.middleware.call('service.restart', 'cron')
         return response
 
+    @accepts(Str('name'), Int('threshold', default=35))
+    async def run(self, name, threshold):
+        """
+        Initiate a scrub of a pool `name` if last scrub was performed more than `threshold` days before.
+        """
+        await self.middleware.call('alert.oneshot_delete', 'ScrubNotStarted', name)
+        await self.middleware.call('alert.oneshot_delete', 'ScrubStarted', name)
+        try:
+            started = await self.__run(name, threshold)
+        except ScrubError as e:
+            await self.middleware.call('alert.oneshot_create', 'ScrubNotStarted', {
+                'pool': name,
+                'text': e.errmsg,
+            })
+        else:
+            if started:
+                await self.middleware.call('alert.oneshot_create', 'ScrubStarted', name)
+
+    async def __run(self, name, threshold):
+        if name == 'freenas-boot':
+            if not await self.middleware.call('system.is_freenas'):
+                if await self.middleware.call('failover.status') == 'BACKUP':
+                    return
```

Comment:
Now that I come to think about this, I cant think of any reason why we wouldnt run scrub of boot pool in standby node.

Reply:
Good that you've noted it, I've put this in a wrong place